blog/2024-09-05-late-chunking/index.mdx

```diff
@@ -0,0 +1,231 @@
+
+![Goldilocks](./img/goldilocks.png)
+
+It appears we have a goldilocks problem, the naive approach may be more cost-effective but can reduce precision, while on the other hand, late interaction and ColBERT offer us increased precision at extreme costs. Surely there must be something in the middle that's just right? Well, late chunking may exactly that.
+
+## How Late Chunking Works
+
+As mentioned earlier late chunking origins are linked closely with late interaction in that both utilise the token-level vector representations that are produced during the forward pass of an embedding model.
+
+Unlike late interaction, there is a pooling step that occurs after the initial inference. This pooling differs from traditional embedding models that pool all representations from every token into a single representation. In late chunking, this pooling is done on segments of the text according to some predetermined chunking strategy that can be aligned based on token spans or boundary cues, thus the term late chunking.
+
+The result is that a long document is still represented by numerous embeddings but critically those embeddings are primed with contextual information relevant to their neighboring chunks.
+
+### What's required
+
+Late chunking requires a relatively simple alteration to the pooling step of the embedding model that can be implemented in **under 30 lines of code** and its vectors can be ingested as individual chunks into a vector database without any modification to the retrieval pipeline.
+
+There are however some requirements needed ahead of performing late chunking:
+
+1. **Long context models** are a requirement as we need token representation for the entirety of the long document to make them contextually aware. Notably, JinaAI tested using their model [jina-embeddings-v2-base-en](https://huggingface.co/jinaai/jina-embeddings-v2-base-en) which has the highest performance to parameter ratio on MTEB's long embed retrieval benchmark. This model supports up to 8192 tokens which is roughly equivalent to 10 standard pages of text. This model also uses a mean pooling strategy in typical behavior which is a requirement for any model looking to take advantage of late interaction.
+
+2. **Chunking logic:** being able to chunk text ahead of inference as well as associating each chunk with its corresponding token spans is also critical to making late chunking work. Luckily there are many ways to create chunks in this manner and given late chunking's ability to condition each chunk on previous ones chunking approaches like fixed-size chunking without any overlap may be all that is needed.
+
+## Late Chunking in Action
+
+In the below head-to-head comparison, we used our recent [blog](https://weaviate.io/blog/launching-into-production) as a sample to test late chunking vs naive chunking against. Using a fixed token chunking strategy (**num tokens = 128**) resulted in the following sentence being split across two different chunks:
```

Comment:
I don't think "against" is needed after the word "native chunking". Though using "vs" is correct, "against" suits this scenario which involves a head-to-head comparison.
```suggestion
In the below head-to-head comparison, we used our recent [blog](https://weaviate.io/blog/launching-into-production) as a sample to test late chunking against naive chunking. Using a fixed token chunking strategy (**num tokens = 128**) resulted in the following sentence being split across two different chunks:
```